controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -303,7 +311,7 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 		}
 
 	} else {
-		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
+		return r.deletionPhase()
```

Comment:
The fact that the client was marked for deletion does not mean we can ignore reconciles going forward. You need to know if the resource is allowed to be deleted or not before deciding if to reconcile or to delete